### DISCUSSION TITLE
Kanban board

### DISCUSSION THREAD

**Discussion by davidak** [elementary/tasks#222] [Ideas]
## Prerequisites
- [X] I have searched open and closed issues for duplicates.

## Feature
**Is your feature request related to a problem? Please describe.**

I'm always frustrated when i feel overwhelmed with the number of tasks i planned to do, especially when i work on so many at once that i couldn't enumerate them. Right now i feel like working on 20-30 projects at once. To-Do lists don't help here and even **Getting things done** (GTD) don't helped me.

**Describe the solution you'd like**

A solution i found yesterday are personal kanban boards. We started to use Kanban at work in our team and it works really well, so i asked myself if that is a common thing to use also for personal tasks.

I found some resources online that this is a thing and really helps people to focus!

https://hackernoon.com/personal-kanban-part-1-why-todo-lists-don-t-work-3b5c6dc78708
http://www.ericlee.info/blog/?p=1068
https://lifehacker.com/productivity-101-how-to-use-personal-kanban-to-visuali-1687948640
https://www.personalkanban.com/personal-kanban-101
https://nomad8.com/articles/everyone-needs-kanbanfor1

So please consider creating a powerful tool that really help people to organize and focus on getting tasks done instead of creating the hundred thousandth basic todo app (there are actually 5 in AppCenter!).

https://appcenter.elementary.io/com.github.alainm23.planner/
https://appcenter.elementary.io/com.github.dahenson.agenda/
https://appcenter.elementary.io/com.github.jmoerman.go-for-it/
https://appcenter.elementary.io/com.github.juarezfranco.tarefas-desktop/
https://appcenter.elementary.io/com.github.lainsce.yishu/

Don't accidentally recreate them!

**Existing work**

https://hackernoon.com/personal-kanban-part-2-personal-kanban-with-trello-a6bd6cdb0588

He seems to really like Trello. But it's a cloud service where you don't own your data. The data should be stored locally in an open format you can migrate to other programs. I really like [todo.txt](http://todotxt.org/) for tasks, but there is no intuitive GUI with kanban support.

**Describe alternatives you've considered**

There are browser-based kanban boards you can host yourself, but no intuitive GUI program that fit's the elementary OS experience.

https://wekan.github.io/
https://kanboard.org/
https://apps.nextcloud.com/apps/deck
https://restya.com/board
https://github.com/huytd/kanelm
https://greggigon.github.io/my-personal-kanban/
https://github.com/kiswa/TaskBoard
https://github.com/apankrat/nullboard
https://github.com/wheatbin/wheatbin

**Comment by alainm23** [comment]
Did you hear about [Firetask](https://firetask.com/)? It's a task manager for macOS that has a Kanban view that gives you an overview of all current tasks and projects. Are you talking about something like that?

**Comment by davidak** [comment]
No, i don't follow the development on macOS anymore. Thanks for the hint.

![Firetask-Mac-3](https://user-images.githubusercontent.com/91113/78525402-57076d80-77d7-11ea-9b88-d0ad91b11773.png)

The design looks clean, but also very limited. It seems you can't configure the colums in their "GTD-optimized kanban board". The second main feature is missing, a **work in progress limit**! So you would only work on for example 3 tasks at a time instead of 30.

You can play around with a Wekan demo here to get a feel:

https://demo.sandstorm.io/appdemo/m86q05rdvj14yvn78ghaxynqz7u2svw6rnttptxx49g1785cdv1h

I consider using the columns **inbox**, **next**, **in progress**, **waiting**, **done.**

![Screenshot from 2020-04-06 07-44-40](https://user-images.githubusercontent.com/91113/78526507-823f8c00-77da-11ea-9ad0-5478de15ef77.png)

You can import this export when creating a new board there: 
[wekan-export-board-sandstorm.json.zip](https://github.com/elementary/tasks/files/4436027/wekan-export-board-sandstorm.json.zip)

**Comment by marbetschar** [comment]
There's quite a lot of statements in your feature request. I try to reflect on as much of this as good as I can.

***

**Disclaimer:** Please be aware this answer contains my personal opinion and experience. It does not reflect an official statement of elementary in any way - but as I contributed quite a bit to the current state of the tasks app and intend to do so in the future, I'd like to share my view on this feature request:

***

> I'm always frustrated when i feel overwhelmed with the number of tasks i planned to do, especially when i work on so many at once that i couldn't enumerate them. Right now i feel like working on 20-30 projects at once. To-Do lists don't help here and even Getting things done (GTD) don't helped me.

I hear you, been there myself: Tried almost every Todo/Productivity Software out there.

**TL;DR**
In my experience, there is no single App which can satisfy the wide range of needs. Further more: I came to the conclusion that successful task management is 80% personal organization and 20% software features. That said, if the App provides the bare minimum of features but does this really well its sufficient (personally I ended up using Apple Reminders when I was on macOS for example).

**The Long Story**
It's a very thin line for an App to provide just enough features to let the user stay productive vs. misleading the user to over-manage his tasks. Personally I ended up to rethinking and slimming down my task-management workflow in an App agnostic way several times blending the approaches of [GTD](https://gettingthingsdone.com/), [Eat the Frog](https://www.fastcompany.com/1592454/work-smart-do-your-worst-task-first-or-eat-live-frog-every-morning), [1-3-5 Rule](https://www.themuse.com/advice/a-better-todo-list-the-135-rule), the [Eisenhower Matrix](https://www.eisenhower.me/eisenhower-matrix/) and others.

I finally ended up with the following system, which primarily targets to build habits to get things done - instead of planning each individual task:

_Task App_
- Catch all new tasks in a Inbox task list
- Use a dedicated task list per project, where "project" is loosely defined and can be anything that takes more effort than one task to complete
- Use a dedicated task list for recurring tasks (mainly for creating new habits e.g.: "Plan next week" every Sunday)
- Use the task reminder feature for time sensitive tasks only (aka "I need to call person A on Monday at 14 o'clock, otherwise he'll be in holiday for two weeks")
- Review all my tasks regularly (ideally every week)

_Calendar App_
- Plan time bound events such as holidays, my weekly gym session, meetings etc.
- Plan time blocks for when I wan't to work on a specific project (see above for a definition of "project") - without actually knowing what I'm exactly doing during this time block
- Always plan a full week ahead (this goes primarily for the project specific time blocks above)

_Daily usage_
As soon as I hit a blocked time in my Calendar I fire up the Task App to figure out what should be done next in my personal projects. I choose the task applying several criteria e.g. which task provides the highest impact right now when done? Is there any time sensitive task which needs to be done as soon as possible? etc.

> A solution i found yesterday are personal kanban boards. We started to use Kanban at work in our team and it works really well, so i asked myself if that is a common thing to use also for personal tasks. I found some resources online that this is a thing and really helps people to focus!

Kanban boards really shine in projects done in a team. I don't really use Kanban for my personal tasks, since when I look at a task I usually already know in what state it is. So to me, the horizontal Kanban UI does not add information for personal projects. If I'd go for a Kanban approach for my personal tasks, I would simply use multiple lists, tags or categories.

Altough, I don't use Kanban you might want to give ClickUp a try - apart from being cloud based, I heard it should be quite good and it supports working in teams as well: https://clickup.com/

> So please consider creating a powerful tool that really help people to organize and focus on getting tasks done instead of creating the hundred thousandth basic todo app (there are actually 5 in AppCenter!). Don't accidentally recreate them!

Actually, we don't accidentally recreate any of these because no one of these support syncing using evolution data server. Which provides a unified approach for synchronization of your tasks with any cloud based service and thus with any end user device you own (iOS, Android, Windows, macOS, Linux).

**To sum all this up**

To conclude, I'm not against an additional Kanban view if there is a broader demand for this feature. But before we start to work on this, we need to focus on getting basic task management done well. You are very welcome to test the current state of work and contribute in any way you like, so we can get there quicker.
